test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -1333,14 +1333,6 @@ private void validate() {
                 position = Position.HORIZONTAL;
             }
 
-            if (panelCreator != null) {
-                if (splitUI && (testWindows != null || windowListCreator != null)) {
-                    // TODO Is it required? We can support both
```

Comment:
@aivanov-jdk please note, I am going to remove this check, it looks artificial to me.

In this test I added a JTextArea to display logs and messages. 
It looks like we could add similar functionality to PassFailJFrame (outside the scope of this issue, of course).

Reply:
This looks like a valid scenario where testUI as well as splitUI are required to be used in combination.

Reply:
This looks like a valid scenario where testUI as well as splitUI are required to be used in combination. Making this change to PassFailJFrame can be helpful.

Reply:
@azvegint, yes, we can support both. But I didn't test it thoroughly, so I decided it was safer to add this.

Please, don't modify `PassFailJFrame` together with tests unless it's strictly required… It is for the sake of facilitating backporting the changes. Otherwise, backporting improvements to the framework will depend on the presence of a test.

In this case, it could be fine though… provided that the test you're updating exists in all the repos.

Either way, I think it's good to create a separate bug for `PassFailJFrame` which you'll resolve with the current one.

Reply:
That's a good point, I created #18281 for that.

Reply:
It's integrated now. Likely, you want to merge master into your branch to avoid possible conflicts, although bot don't complain about any conflict.